python/pyspark/sql/dataframe.py

```diff
@@ -1223,10 +1226,8 @@ def withColumnRenamed(self, existing, new):
         >>> df.withColumnRenamed('age', 'age2').collect()
         [Row(age2=2, name=u'Alice'), Row(age2=5, name=u'Bob')]
         """
-        cols = [Column(_to_java_column(c)).alias(new)
-                if c == existing else c
-                for c in self.columns]
-        return self.select(*cols)
+        assert existing in self.columns, "%s is not an existing column" % existing
```

Comment:
Good catch, will remove it. I'm really surprised by the behavior in Scala.

The reason we want to have some check on Python side is that the Java exception is not easy  to understand for Python programmer (nested inside a Py4j exception). The Python exception or messages does improve the experience for Python programmer, especially beginners.